tests/functional/conftest.py

```diff
@@ -48,10 +55,31 @@
 # Directory storing test-specific data.
 _DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
 
-# Provide test access to _DATA_DIR via a fixture.
+# Code
+# ====
+# Fixtures
+# --------
+# Define a fixure for the DataDir object.
 @pytest.fixture
-def data_dir():
-    return _DATA_DIR
+def data_dir(
+  # The request object for this test. See
+  # https://pytest.org/latest/builtin.html#_pytest.python.FixtureRequest
+  # and
+  # https://pytest.org/latest/fixture.html#fixtures-can-introspect-the-requesting-test-context.
+  request,
+  # The tmpdir object for this test. See
+  # https://pytest.org/latest/tmpdir.html.
+  tmpdir):
+
+    # Strip the leading 'test_' from the test's name.
+    name = request.function.__name__[5:]
```

Comment:
@codewarrior0 The namespace-tests in the old test-suite (test_nspkg1*, test_nspkg1*) share the same packages and modules. But maybe it's simpler to put them into a test-module (e.g. `test_namespaces`) and define the fixture there (with `scope='module'`). If we find this to be used more often, we can generalize it later on.